src/main/java/io/fabric8/vertx/maven/plugin/utils/FileUtils.java

```diff
@@ -15,6 +18,21 @@
  */
 public class FileUtils {
 
+
+    public static Set<Path> includedDirs(MavenProject project, Optional<List<String>> includes) {
+
+        //TODO Handle resource resource roots and compile roots from respective plugins and includes
```

Comment:
yes this done now, now all compile source roots and resource roots are watched by the Incremental Builder